From 8827eac24deaa89778d8c133f002f921f13e61cd Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Wed, 1 Jul 2020 04:47:28 +0530
Subject: [PATCH 1407/1851] spi: spi-xilinx: Fix type mismatch in
 xilinx_spi_irq API

commit 23fc53dcd3ff8d610edca4e5517f972746f2ef22 from
https://github.com/Xilinx/linux-xlnx.git

There was a mismatch between the return type of "xilinx_spi_irq"
API(irqreturn_t) and the type of the variable(int) that the API
was returning. Updated return variable type from "int" to
"irqreturn_t".

Addresses-Coverity: "mixed_enum_type"
Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Acked-by: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/spi/spi-xilinx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-xilinx.c b/drivers/spi/spi-xilinx.c
index 33818b7b09a1..8c9a28a286b3 100644
--- a/drivers/spi/spi-xilinx.c
+++ b/drivers/spi/spi-xilinx.c
@@ -615,7 +615,7 @@ static irqreturn_t xilinx_spi_irq(int irq, void *dev_id)
 	struct spi_master *master = dev_id;
 	struct xilinx_spi *xspi = spi_master_get_devdata(dev_id);
 	u32 ipif_isr;
-	int status = IRQ_NONE;
+	irqreturn_t status = IRQ_NONE;
 
 	/* Get the IPIF interrupts, and clear them immediately */
 	ipif_isr = xspi->read_fn(xspi->regs + XIPIF_V123B_IISR_OFFSET);
-- 
2.31.1

